pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -387,18 +389,22 @@ func (o *OpenIDCProvider) getUserInfo(ctx *context.Context, config *v32.OIDCConf
 	}
 	// Valid will return false if access token is expired
 	if !oauth2Token.Valid() {
-		// since token is not valid, the TokenSource func will attempt to refresh the access token
+		// since token is not valid, the TokenSource func used in the Client func will attempt to refresh the access token
 		// if the refresh token has not expired
-		logrus.Debugf("[generic oidc] saveOIDCConfig: attempting to refresh access token")
+		logrus.Debugf("[generic oidc] getUserInfo: attempting to refresh access token")
 	}
 	logrus.Debugf("[generic oidc] saveOIDCConfig: getting user info")
+	oldOauth2Token = oauth2Token
 	userInfo, err = provider.UserInfo(updatedContext, oauthConfig.TokenSource(updatedContext, oauth2Token))
 	if err != nil {
 		return userInfo, oauth2Token, err
 	}
 	if err := userInfo.Claims(&claimInfo); err != nil {
 		return userInfo, oauth2Token, err
 	}
+	if reflect.DeepEqual(oauth2Token, oldOauth2Token) {
+		o.UpdateToken(oauth2Token, claimInfo.Name)
+	}
```

Comment:
Should the token be updated if not equal?

Reply:
You are correct - I have updated this in my most recent changes.